src/Microsoft.OData.Core/PropertyCache.cs

```diff
@@ -21,8 +21,6 @@ public PropertySerializationInfo GetProperty(string name, string uniqueName, IEd
             PropertySerializationInfo propertyInfo;
             if (!propertyInfoDictionary.TryGetValue(uniqueName, out propertyInfo))
             {
-                WriterValidationUtils.ValidatePropertyName(name);

```

Comment:
Just remove the validate?  or use the validate setting?

Reply:
This is only called from WriteProperty.  By moving the ValidatePropertyName first in WriteProperty, we do the validation (correctly checking whether or not the type is open) before this code ever gets called.  Which is good, since this code doesn't have the context of the defining type to know whether or not it was open.

---
In reply to: [154829612](https://github.com/OData/odata.net/pull/1010#discussion_r154829612) [](ancestors = 154829612)